File: Lab1/Exercise2/MainActivity.java
package com.example.mainactivity;

import android.os.Bundle;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    TextView tvTitle, tvName;
    EditText etName;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        tvTitle = findViewById(R.id.tvTitle);
        tvName = findViewById(R.id.tvName);
        etName = findViewById(R.id.etName);

        etName.setOnFocusChangeListener((v, hasFocus) -> {
            if (!hasFocus) { // when user finishes typing
                String name = etName.getText().toString();
                if (!name.isEmpty()) {
                    tvTitle.setText("Welcome, " + name + "!");
                } else {
                    tvTitle.setText("User Information");
                }
            }
        });
    }
}

File: Lab1/Exercise4/MainActivity.kt
package com.example.mainactivity

import android.os.Bundle
import android.widget.Button
import android.widget.ImageButton
import android.widget.Toast
import android.widget.ToggleButton
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        val buttonStandard: Button = findViewById(R.id.buttonStandard)
        val buttonImage: ImageButton = findViewById(R.id.buttonImage)
        val toggleButton: ToggleButton = findViewById(R.id.toggleButton)

        buttonStandard.setOnClickListener {
            Toast.makeText(this, "Scanning barcode...", Toast.LENGTH_SHORT).show()
        }

        buttonImage.setOnClickListener {
            Toast.makeText(this, "Barcode image clicked!", Toast.LENGTH_SHORT).show()
        }

        toggleButton.setOnCheckedChangeListener { _, isChecked ->
            val message = if (isChecked) "Scanner is now enabled." else "Scanner disabled."
            Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
        }
    }
}
